File: Wed_Thời Trang Nam/ThuanBackend/Thuan.API/Models/UserBehavior.cs
// Models/UserBehavior.cs
using System.ComponentModel.DataAnnotations;

namespace Thuan.API.Models
{
    public class UserBehavior
    {
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; } // ID người dùng (hoặc 0 nếu không login)

        [Required]
        public string EventType { get; set; } = string.Empty; // "view", "click", "add_to_cart", "search", "purchase"

        public int? ProductId { get; set; } // ID sản phẩm (nếu có liên quan)

        public string EventData { get; set; } = string.Empty; // JSON data bổ sung

        // === TIME TRACKING (MỚI) ===
        public int? DurationSeconds { get; set; } // Bao lâu xem (giây)

        public string? PageName { get; set; } // Trang nào: "index", "sanpham", "chitietsp", ...

        public string? Interest { get; set; } // Quan tâm gì: JSON {"color": "đỏ", "style": "casual", "size": "M"}

        public DateTime? StartedAt { get; set; } // Lúc nào bắt đầu

        public DateTime? EndedAt { get; set; } // Lúc nào kết thúc

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}

File: Wed_Thời Trang Nam/ThuanBackendNode/models/UserBehavior.js
// models/UserBehavior.js
const mongoose = require('mongoose');

const UserBehaviorSchema = new mongoose.Schema({
    userId: { type: Number, required: true }, // ID from MySQL users table
    eventType: { type: String, required: true }, // "view", "click", "add_to_cart", "search", "purchase", "scroll"
    productId: { type: Number, default: null }, // ID from MySQL products
    eventData: { type: String, default: '' },
    durationSeconds: { type: Number, default: null },
    pageName: { type: String, default: null },
    interest: { type: String, default: null }, // JSON or string
    startedAt: { type: Date, default: null },
    endedAt: { type: Date, default: null },
    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('UserBehavior', UserBehaviorSchema);
